### PR TITLE
Check for null responses during cosmos updates to avoid NPE

### PR DESCRIPTION
**Pull request by Arun-LinkedIn**
cosmosDataAccessor.updateMetadata() can return null response if there is nothing to update, i.e. fields already have specified value. Check for null response during metadata updates to avoid NPE.

### CODE REVIEW

File: ambry-cloud/src/main/java/com/github/ambry/cloud/azure/AzureBlobDataAccessor.java [hunk 1]
```diff
@@ -335,7 +335,7 @@ public CloudBlobMetadata getBlobMetadata(BlobId blobId) throws BlobStorageExcept
    * @throws IllegalStateException on request timeout.
    */
   public AzureCloudDestination.UpdateResponse updateBlobMetadata(BlobId blobId, Map<String, Object> updateFields,
-      CloudUpdateValidator cloudUpdateValidator) throws BlobStorageException {
+      CloudUpdateValidator cloudUpdateValidator) throws Exception {
```
Comment (ankagrawal) [inline ambry-cloud/src/main/java/com/github/ambry/cloud/azure/AzureBlobDataAccessor.java:338]: Why do we need this change? We are either throwing BlobStorageException here, or RunTimeException.

File: ambry-cloud/src/main/java/com/github/ambry/cloud/azure/AzureCloudDestination.java [hunk 2]
```diff
@@ -399,7 +399,9 @@ private UpdateResponse updateBlobMetadata(BlobId blobId, Map<String, Object> upd
       boolean updatedCosmos;
       try {
         CosmosItemResponse<CloudBlobMetadata> response = cosmosDataAccessor.updateMetadata(blobId, metadataMap);
-        updatedCosmos = response.getItem() != null;
+        // cosmosDataAccessor.updateMetadata() can return null response if there is nothing to update, i.e. fields
+        // already have specified value. Check for null response to see if metadata was updated.
+        updatedCosmos = response != null;
```
Comment (ankagrawal) [inline ambry-cloud/src/main/java/com/github/ambry/cloud/azure/AzureCloudDestination.java:404]: No integration test needed for this?

Comment (SophieGuo410) [inline ambry-cloud/src/main/java/com/github/ambry/cloud/azure/AzureCloudDestination.java:404]: if response != null, response.getItem() can still be null If responseBodyAsByteArray is empty?

Comment (Arun-LinkedIn) [reply ambry-cloud/src/main/java/com/github/ambry/cloud/azure/AzureCloudDestination.java]: I was able to reproduce this with existing integration tests  (like [this](https://github.com/linkedin/ambry/blob/master/ambry-cloud/src/test/java/com/github/ambry/cloud/CloudBlobStoreIntegrationTest.java#L229) )  where there is no changes in metadata fields to update. Verified that we are not hitting NPE after the fix.

File: ambry-cloud/src/main/java/com/github/ambry/cloud/azure/AzureBlobDataAccessor.java (hunk 1 above)
Comment (Arun-LinkedIn) [reply ambry-cloud/src/main/java/com/github/ambry/cloud/azure/AzureBlobDataAccessor.java]: Looks like we could also throw `StoreException` during validation of update method when we call `cloudUpdateValidator.validateUpdate` within this function. Previously, `StoreException` was also incorrectly wrapped in `RunTimeException` which seemed incorrect. Due to that, I had to modify the signature of the method to throw `StoreException` explicitly.

**Review by SophieGuo410 — APPROVED** [review]

**Review by ankagrawal — APPROVED** [review]

**Pull request merged**